test_util/provide_aws.py

```diff
@@ -0,0 +1,215 @@
+        reservation = self.ec2_client.describe_instances(Filters=[{
+                'Name': 'tag-value', 'Values': [self.stack_name]}])['Reservations']
+        for inst_i in reservation[0]['Instances']:
+            inst = self.session.resource('ec2', region_name=self.region).Instance(inst_i['InstanceId'])
+            inst.create_tags(
+                    DryRun=False,
+                    Tags=[
+                        {
+                            'Key': 'expiration',
+                            'Value': expiration_string
+                        },
+                        {
+                            'Key': 'owner',
+                            'Value': '#dcos-ci'
+                        }])
+
+    def create_key_pair(self, key_name):
+        """
+        Args:
+            key_name (str): key name to be registered in AWS
+        Returns:
+            string of contents of key
+        """
+        key = self.ec2_client.create_key_pair(KeyName=key_name)
+        return key['KeyMaterial']
+
+    def delete_key_pair(self, key_name):
+        self.ec2_client.delete_key_pair(KeyName=key_name)
+
+    @retrying.retry(stop_max_delay=1200*1000,
+                    wait_fixed=3000,
+                    retry_on_exception=lambda ex: False,
+                    retry_on_result=lambda res: res is False)
+    def wait_for_cluster_creation(self):
+        logger.info('Waiting for cluster to be created...')
+        self.stack.reload()
+        current_status = self.stack.stack_status
+        if current_status == 'CREATE_IN_PROGRESS':
+            return False
+        if current_status != 'CREATE_COMPLETE':
+            raise Exception("Expected stack {} to have status CREATE_COMPLETE,"
+                            "but found: {}".format(self.stack.stack_id,
+                                                   current_status))
+
+    @retrying.retry(stop_max_delay=1200*1000,
+                    wait_fixed=3000,
+                    retry_on_exception=lambda ex: False,
+                    retry_on_result=lambda res: res is False)
+    def wait_for_cluster_deletion(self):
+        logging.info('Waiting for cluster to be deleted...')
+        current_status = self.stack.stack_status
+        if current_status == 'DELETE_IN_PROGRESS':
+            return False
+        try:
+            self.stack.reload()
+        except botocore.exceptions.ClientError as ex:
+            resp = ex.response['Error']
+            if (
+                    resp['Error']['Code'] == 'ValidationError' and
+                    resp['Error']['Message'].endswith('does not exist')):
+                logging.info("Boto cannot find stack; deletion successful")
+                return True
+            else:
+                raise ex
+        if current_status != 'DELETE_COMPLETE':
+            raise Exception('Delete finished, but cluster has unexpected state: '+current_status)
+
+
+class CloudFormation(AwsProvision):
+    def create(self, name, template_url,
```

Comment:
I don't think this is quite how we want to structure this. I think it would be better to have a generic CloudFormation provisioner that can do stack standup / shutdown with an arbitrary template url + python format dictionary of parameters, then we use / call that for the AWS Simple, advanced, etc. stacks that way we don't have to do things like the simple python doct parameters -> complicated AWS ParameterKey + ParameterValue in multiple places.